src/executables/csql.c

```diff
@@ -3331,3 +3362,133 @@ get_host_ip (unsigned char *ip_addr)
 
   return 0;
 }
+
+static int
+csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
+{
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char buf[DB_MAX_USER_LENGTH + DB_NAME_LEN + CUB_MAXHOSTNAMELEN + 1] = { 0, };
+  const char *delim = " \n";
+  char *save_ptr_strtok = NULL;
+  const char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
+  char *p = NULL;
+  const char *err_msg;
+  CSQL_ARGUMENT csql_new_arg;
+
+  if (argument == NULL)
+    {
+      err_msg = (*csql_get_message) (CSQL_MSG_TOO_FEW_ARGS);
+      fprintf (stderr, err_msg);
+      return DO_CMD_SUCCESS;
+    }
+
+  memset (&csql_new_arg, 0, sizeof csql_new_arg);
+
+  strncpy (buf, argument, sizeof (buf) - 1);
+
+  if ((user_name_ptr = strtok_r (buf, delim, &save_ptr_strtok)) == NULL)
+    {
+      csql_Error_code = CSQL_ERR_SQL_ERROR;
+      return DO_CMD_FAILURE;
+    }
+
+  /*find db name following the user name */
+  db_name_ptr = strtok_r (NULL, delim, &save_ptr_strtok);
+  if (db_name_ptr == NULL)
+    {
+      db_name_ptr = csql_arg->db_name;
+    }
+  else
+    {
+      if (csql_arg->sa_mode == true)
+	{
+	  fprintf (csql_Output_fp, "CONNECT session command allows the login with only user name in the --SA-mode.\n");
```

Comment:
Just opinion,
`fprintf (csql_Output_fp, "Cannot connect to another DB in the SA mode.\n");`

Reply:
Thank you. 
I will apply as "Cannot connect to other DB in the SA mode."